keg_auth/libs/authenticators.py

```diff
@@ -296,19 +431,88 @@ def on_form_valid(self, form):
         try:
             user = self.parent.verify_user(login_id=form.login_id.data, password=form.password.data)
 
+            if self.should_limit_attempts():
```

Comment:
If attempts are being blocked, we want to block it before the call to `verify_user`. When evaluating for blockage, we shouldn't care if the input is valid or not - just whether the request can proceed, and log the attempt regardless.